tests/test_sensor.py

```
@@ -0,0 +1,272 @@
+            CONF_VARIABLE_ID: "string_value",
+            CONF_VALUE: "unchanged",
+            "value_type": "string",
+            CONF_YAML_VARIABLE: False,
+            CONF_RESTORE: False,
+            CONF_FORCE_UPDATE: False,
+        }
+    )
+    assert await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    with pytest.raises(ValueError, match="Cannot .* non-numeric variable"):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)